titan/features/incar.py

```diff
@@ -0,0 +1,202 @@
+from typing import Dict, ClassVar, Set
+
+from . import base_feature
+from .. import utils
+from .. import agent
+from .. import population
+from .. import model
+
+
+class Incar(base_feature.BaseFeature):
+
+    name = "incar"
+    stats = ["incar", "incar_hiv", "new_release", "new_release_hiv"]
+    """
+        Incar collects the following stats:
+
+        * incar - number of agents with active incar
+        * incar_hiv - number of agents with active incar and HIV
+        * new_release - number of agents released this timestep
+        * new_release_hiv - number of agents releasted this timestep with HIV
+    """
+
+    new_releases: ClassVar[Set["agent.Agent"]] = set()
+
+    def __init__(self, agent: "agent.Agent"):
+        super().__init__(agent)
+
+        self.active = False
+        self.time = 0
+
+    def init_agent(self, pop: "population.Population", time: int):
+        """
+        Initialize the agent for this feature during population initialization (`Population.create_agent`).  Called on only features that are enabled per the params.
+
+        Run incarceration assignment on an agent.  The duration of incarceration at initialization is different than the ongoing to reflect that agents with longer durations will be more highly represented in that population at any given point in time.
+
+        args:
+            pop: the population this agent is a part of
+            time: the current time step
+        """
+        incar_params = self.agent.location.params.demographics[self.agent.race][
+            self.agent.sex_type
+        ].incar
+        jail_duration = incar_params.duration.init
+
+        prob_incar = incar_params.init
+        if pop.pop_random.random() < prob_incar:
+            self.active = True
+            bin = 1
+            current_p_value = jail_duration[bin].prob
+            p = pop.pop_random.random()
+
+            while p > current_p_value:
+                bin += 1
+                current_p_value += jail_duration[bin].prob
+
+            self.time = pop.pop_random.randrange(
+                jail_duration[bin].min, jail_duration[bin].max
+            )
+
+    def update_agent(self, model: "model.HIVModel"):
+        """
+        Update the agent for this feature for a time step.  Called once per time step in `HIVModel.update_all_agents`. Agent level updates are done after population level updates.   Called on only features that are enabled per the params.
+
+        Incarcerate an agent or update their incarceration variables
+
+        args:
+            model: the instance of HIVModel currently being run
+        """
+        hiv_bool = self.agent.hiv
+
+        if hiv_bool:
+            hiv_multiplier = self.agent.location.params.incar.hiv.multiplier
+        else:
+            hiv_multiplier = 1
```

Comment:
```suggestion
            hiv_multiplier = 1.0
```
Just to make it explicit that this is a float.